packages/flutter/test/material/app_bar_theme_test.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
```

Comment:
It would be good to add a defensive regression test for the hashCode,copyWith, == methods. Something like:
```
  test('AppBarTheme copyWith operator ==, hashCode basics()', () {
    expect(const AppBarTheme(), const AppBarTheme().copyWith()));
    expect(const TextTheme().hashCode, const AppBarTheme().copyWith().hashCode));
  });
```

Reply:
Good idea, done

Reply:
Let's make that a best practice for all of us. Please put it in the coding traditions section of our team doc.